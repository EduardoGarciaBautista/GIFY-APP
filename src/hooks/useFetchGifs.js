import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";


export const useFetchGifs = (category) => {
    const [state, setstate] = useState({
        data: [],
        loading: true,
    });

    useEffect(() => {
        getGifs(category).then(result => setstate({
            data: result,
            loading: false
        }));
    }, [category]);

    return state;
}