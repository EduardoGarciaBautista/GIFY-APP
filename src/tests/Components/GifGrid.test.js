import React from 'react';
import '@testing-library/jest-dom';
import {shallow}  from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
const { default: GifGrid } = require("../../Components/GifGrid");

jest.mock('../../hooks/useFetchGifs');



describe('test for GifGrid', () => {
   
    const category = 'MOUSE';
    


    test('should component be equal to snapshot', () => {
       
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();

    });
    

    test('should show item when images are loaded', () => {

        const gifs = [{
            id: 'fdgfsd',
            url: 'https://test.com',
            title: 'Test Mock'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('p').exists()).toBe(false);

        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
   
    });
    

});