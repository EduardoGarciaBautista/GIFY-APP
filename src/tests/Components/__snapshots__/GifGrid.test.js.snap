// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for GifGrid should component be equal to snapshot 1`] = `
<div
  className="card-grid"
>
  <h1>
    Cargando...
  </h1>
</div>
`;

exports[`test for GifGrid should show item when images are loaded 1`] = `
<div
  className="card-grid"
>
  <GifGridItem
    id="fdgfsd"
    key="fdgfsd"
    title="Test Mock"
    url="https://test.com"
  />
</div>
`;
